volmdlr/faces.py

```diff
@@ -2710,19 +2720,200 @@ def point2d_with_dimension_to_3d(self, point2d, points_x, points_y, xmin, xmax,
         return self.point2d_to_3d(point2d_01)
     
 
+    def linesegment2d_parametric_to_dimension(self, linesegment2d, points_x, points_y):
+        '''
+        convert a linesegment2d from the parametric to the dimensioned frame
+        '''
+
+        xmin, xmax, ymin, ymax = 0, 1, 0, 1
+
+        linesegment2d_with_dimension = volmdlr.edges.LineSegment2D(self.point2d_parametric_to_dimension(linesegment2d.start, points_x, points_y, xmin, xmax, ymin, ymax), self.point2d_parametric_to_dimension(linesegment2d.end, points_x, points_y, xmin, xmax, ymin, ymax))
+
+        return linesegment2d_with_dimension
+
+
+    def linesegment3d_to_2d_with_dimension(self, linesegment3d, points_x, points_y):
+        '''
+        compute the linesegment2d of a linesegment3d, on a Bspline surface, in the dimensioned frame
+        '''
+
+        linesegment2d = self.linesegment3d_to_2d(linesegment3d)
+        linesegment2d_with_dimension = self.linesegment2d_parametric_to_dimension(linesegment2d, points_x, points_y)
+
+        return linesegment2d_with_dimension
+
+
+    def linesegment2d_with_dimension_to_parametric_frame(self, linesegment2d):            
+        '''
+        convert a linesegment2d from the dimensioned to the parametric frame
+        '''
+
+        for key in self._grids2d:
+            [points_x, points_y, xmin, xmax, ymin, ymax] = key
```

Comment:
You are overwriting points_x, points_y, xmin, xmax, ymin, ymax each time you iterate on the loop. So it turns out to be equal to `[points_x, points_y, xmin, xmax, ymin, ymax] = self._grids2d[-1]`